src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCommandCommand.java

```diff
@@ -177,84 +127,56 @@ public static Command matchCommandByName(ScriptEntry scriptEntry, ElementTag nam
-                Role role = DiscordRoleTag.valueOf(item, scriptEntry.getContext()).role;
-                if (enable) {
-                    result = CommandPrivilege.enable(role);
-                }
-            }
-            if (result != null) {
-                privileges.add(result);
-            }
-            else {
-                Debug.echoError("Privileged input must be a DiscordUserTag or DiscordRoleTag!");
-            }
-        }
-    }
-
-    @Override
-    public void execute(ScriptEntry scriptEntry) {
-        DiscordBotTag bot = scriptEntry.requiredArgForPrefix("id", DiscordBotTag.class);
-        ElementTag commandInstruction = scriptEntry.getElement("instruction");
-        DiscordGroupTag group = scriptEntry.getObjectTag("group");
-        ElementTag name = scriptEntry.getElement("name");
-        ElementTag description = scriptEntry.getElement("description");
-        MapTag options = scriptEntry.getObjectTag("options");
-        ElementTag enabled = scriptEntry.getElement("enabled");
-        ListTag enableFor = scriptEntry.getObjectTag("enable_for");
-        ListTag disableFor = scriptEntry.getObjectTag("disable_for");
-        ElementTag type = scriptEntry.argForPrefixAsElement("type", "slash");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), bot, commandInstruction, group, name, description, options, enabled, enableFor, disableFor, type);
-        }
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgPrefixed @ArgName("id") DiscordBotTag bot,
+                                   @ArgName("instruction") DiscordCommandInstruction instruction,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("group") DiscordGroupTag group,
+                                   @ArgPrefixed @ArgName("name") String name,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("type") Command.Type type,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("description") String description,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("options") MapTag options,
+                                   // Past-deprecated arguments
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("enabled") ElementTag enabled,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("enable_for") ListTag enableFor,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("disable_for") ListTag disableFor) {
         if (group != null && group.bot == null) {
             group.bot = bot.bot;
         }
         JDA client = bot.getConnection().client;
         Bukkit.getScheduler().runTaskAsynchronously(DenizenDiscordBot.instance, () -> {
             try {
-                if (commandInstruction == null) {
+                Command.Type commandType = type != null ? type : Command.Type.SLASH;
```

Comment:
If there should be a default, just set that default in the autoExecute param data